### PR TITLE
feat(db): private settings are merged into metadata

### LINKED ISSUE #13881: [Feature]: Move private settings into metadata
### Describe the feature

Drop the private settings table (and all related API) and move the values into metadata. This allows us to make better use of the value_types in metadata. Also makes it possible to store arrays of values. 

Metadata and Private Settings table structure is almost identical at the moment.

### Is your feature request related to a problem?

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

https://github.com/Elgg/Elgg/issues/6432

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #13881

### CODE REVIEW

**Review by jeabakker — CHANGES_REQUESTED** [review]

File: engine/classes/Elgg/Traits/Entity/PluginSettings.php [hunk 1]
```diff
@@ -33,22 +33,22 @@ public function setPluginSetting(string $plugin_id, string $name, $value): bool
 		
 		$name = $this->getNamespacedPluginSettingName($plugin_id, $name);
 		
-		return $this->setPrivateSetting($name, $value);
+		return $this->setMetadata($name, $value);
```
[the remark this answers is not in the document]
Comment (jdalsem) [reply engine/classes/Elgg/Traits/Entity/PluginSettings.php:36]: nope need to return a status

File: engine/classes/Elgg/Traits/Entity/PluginSettings.php [hunk 2]
```diff
@@ -33,22 +33,22 @@ public function setPluginSetting(string $plugin_id, string $name, $value): bool
...
 	 *
-	 * @return string
+	 * @return mixed
 	 */
-	public function getPluginSetting(string $plugin_id, string $name, $default = null): string {
+	public function getPluginSetting(string $plugin_id, string $name, $default = null) {
 		$name = $this->getNamespacedPluginSettingName($plugin_id, $name);
 		
-		return (string) ($this->getPrivateSetting($name) ?? $default);
+		return $this->getMetadata($name) ?? $default;
```
[the remark this answers is not in the document]
Comment (jdalsem) [reply engine/classes/Elgg/Traits/Entity/PluginSettings.php:51]: nope need to return a status

File: engine/classes/Elgg/Traits/Entity/PluginSettings.php [hunk 3]
```diff
@@ -62,11 +62,11 @@ public function getPluginSetting(string $plugin_id, string $name, $default = nul
 	public function removePluginSetting(string $plugin_id, string $name): bool {
 		$name = $this->getNamespacedPluginSettingName($plugin_id, $name);
 		
-		return $this->removePrivateSetting($name);
+		return $this->deleteMetadata($name);
```
[the remark this answers is not in the document]
Comment (jdalsem) [reply engine/classes/Elgg/Traits/Entity/PluginSettings.php:65]: nope need to return a status

**Comment by jdalsem** [comment]
Replaced by #14143

**Pull request closed without merging**